src/common/tests_dmg_helpers.c

```diff
@@ -0,0 +1,388 @@
+}
+
+static char *
+rank_list_to_string(const d_rank_list_t *rank_list)
+{
+	char		*ranks_str = NULL;
+	int		 width;
+	int		 i;
+	int		 idx = 0;
+
+	if (rank_list == NULL)
+		return NULL;
+
+	width = 0;
+	for (i = 0; i < rank_list->rl_nr; i++)
+		width += snprintf(NULL, 0, "%d,", rank_list->rl_ranks[i]);
+	width++;
+	D_ALLOC(ranks_str, width);
+	if (ranks_str == NULL)
+		return NULL;
+	for (i = 0; i < rank_list->rl_nr; i++)
+		idx += sprintf(&ranks_str[idx], "%d,", rank_list->rl_ranks[i]);
+	ranks_str[width - 1] = '\0';
+
+	return ranks_str;
+}
+
+int
+dmg_pool_create(const char *dmg_config_file,
+		uid_t uid, gid_t gid, const char *grp,
+		const d_rank_list_t *tgts,
+		daos_size_t scm_size, daos_size_t nvme_size,
+		d_rank_list_t *svc, uuid_t uuid)
+{
+	int			argcount = 0;
+	char			**args = NULL;
+	daos_mgmt_pool_info_t	pool_info = {};
+	struct json_object	*dmg_out = NULL;
+	int			rc = 0;
+
+	if (grp != NULL) {
+		args = cmd_push_arg(args, &argcount,
+				"--sys=%s ", grp);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	if (tgts != NULL) {
+		char *ranks_str = rank_list_to_string(tgts);
+		if (ranks_str == NULL) {
+			D_ERROR("failed to create rank string\n");
+			D_GOTO(out_cmd, rc = -DER_NOMEM);
+		}
+		args = cmd_push_arg(args, &argcount,
+				"--ranks=%s ", ranks_str);
+		D_FREE(ranks_str);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	args = cmd_push_arg(args, &argcount,
+			    "--user=%d ", uid);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	args = cmd_push_arg(args, &argcount,
+			    "--group=%d ", gid);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	args = cmd_push_arg(args, &argcount,
+			    "--scm-size=%"PRIu64"b ", scm_size);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	if (nvme_size > 0) {
+		args = cmd_push_arg(args, &argcount,
+				    "--nvme-size=%"PRIu64"b ", nvme_size);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	rc = daos_dmg_json_pipe("pool create", dmg_config_file,
+				args, argcount, &dmg_out);
+	if (rc != 0) {
+		D_ERROR("dmg failed");
+		goto out_json;
+	}
+
+	rc = parse_pool_info(dmg_out, &pool_info);
+	if (rc != 0) {
+		D_ERROR("failed to parse pool info\n");
+		goto out_json;
+	}
+
+	uuid_copy(uuid, pool_info.mgpi_uuid);
+	if (svc == NULL)
+		goto out_svc;
+
+	rc = d_rank_list_dup(&svc, pool_info.mgpi_svc);
+	if (rc != 0) {
+		D_ERROR("failed to dup svc rank list\n");
+		goto out_svc;
+	}
+
+out_svc:
+	d_rank_list_free(pool_info.mgpi_svc);
+out_json:
+	if (dmg_out != NULL)
+		json_object_put(dmg_out);
+out_cmd:
+	cmd_free_args(args, argcount);
+out:
+	return rc;
+}
+
+int
+dmg_pool_destroy(const char *dmg_config_file,
+		 const uuid_t uuid, const char *grp, int force)
+{
+	char			uuid_str[DAOS_UUID_STR_SIZE];
+	int 			argcount = 0;
+	char 			**args = NULL;
+	struct json_object	*dmg_out = NULL;
+	int			rc = 0;
+
+	uuid_unparse_lower(uuid, uuid_str);
+	args = cmd_push_arg(args, &argcount,
+ 			    "--pool=%s ", uuid_str);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs